src/app/components/challenge/challengeleaderboard/challengeleaderboard.component.ts

```diff
@@ -404,6 +422,42 @@ export class ChallengeleaderboardComponent implements OnInit, AfterViewInit {
       data => {
         SELF.updateLeaderboardResults(data['results'], SELF);
         SELF.startLeaderboard(SELF.selectedPhaseSplit['id']);
+        SELF.sortLeaderboardTextOption = SELF.showLeaderboardByLatest ? 'Sort by best' : 'Sort by latest';
```

Comment:
How about instead of making any change here on `refreshLeaderboard` function we update `sortLeaderboardTextOption` after success API call of `showLeaderboardByLatestOrBest` function?

Reply:
Changed!